5m5v-bot.js

```diff
@@ -1,97 +1,115 @@
-  if (user.apiKey) {
-    return user.apiKey;
-  }
+const languageKeys = {
+  english: 'en',
+  french: 'fr',
+  spanish: 'es',
+  german: 'de',
+};
 
-  while (true) {
-    try {
-      console.log(`Logging in as ${user.username}...`);
-
-      await new Promise(resolve => setTimeout(resolve, loginDelayMs));
-
-      user.apiKey = await new Rettiwt().auth.login(user.email, user.username, user.password);
+const tweetFilter = new TweetFilter([], Object.keys(languageKeys));
 
-      console.log('Logged in!');
-
-      return user.apiKey;
-    } catch (e) {
-      console.error(`Unable to log in: ${e.message}`);
-
-      if (!retry) {
-        throw e;
-      }
-    }
-  }
-}
+const streamFilter = {
+  includeWords: [util.trackedTerms.map(term => `"${term}"`).join(' OR ')],
+};
 
 (async () => {
   while (true) {
-    const rettiwt = new Rettiwt({ apiKey: await getApiKey(config.users[0], { retry: true }) });
+    const rettiwt = new Rettiwt({ apiKey: await login() });
 
     console.log(isDryRun ? 'Looking for new tweets (dry run)...' : 'Looking for new tweets...');
 
     try {
-      for await (const tweet of rettiwt.tweet.stream({ includeWords: [util.trackedTerms.map(term => `"${term}"`).join(' OR ')] }, pollingIntervalMs)) {
+      for await (const tweet of rettiwt.tweet.stream(streamFilter, pollingIntervalMs)) {
         const matchingLanguages = tweetFilter.matches(tweet) || [];
 
         for (const language of matchingLanguages) {
-          await new Promise(resolve => setTimeout(resolve, retweetDelayMs));
-
-          const user = config.users.find(user => user.language === language);
-
           try {
             if (!isDryRun) {
-              const rettiwt = new Rettiwt({ apiKey: await getApiKey(user, { retry: false }) });
-
-              await rettiwt.tweet.retweet(tweet.id);
+              await axios.post(process.env.TWEETS_API_ENDPOINT, {
+                lang: languageKeys[language],
+                tweets: [buildTweetPayload(tweet)],
+              }, {
+                headers: {
+                  'X-API-KEY': process.env.TWEETS_API_KEY,
+                },
+             });
             }
 
-            console.log(`Retweeted tweet ${tweet.id} in ${language}:\n${tweet.fullText}`);
+            console.log(`Sent tweet ${tweet.id} in ${language}:\n${tweet.fullText}`);
           } catch (error) {
-            console.error(`Unable to retweet ${tweet.id} in ${language}: ${error.message}`);
-
-            if (error.constructor.name === 'RettiwtError' && error.code === 32) {
-              user.apiKey = null;
-            }
+            console.error(`Unable to send tweet ${tweet.id} in ${language}: ${error.message}`);
           }
         }
       }
     } catch (error) {
       console.error(`Error while streaming tweets: ${error.message}`);
 
       if (error.constructor.name === 'RettiwtError' && error.code === 32) {
-        config.users[0].apiKey = null;
+        await login();
```

Comment:
Probably not needed  as the login is performed on top of the loop

Reply:
Should make more sense now (the loop will no longer log in by default).